proto/frequenz/dispatch/v1/dispatch.proto

```diff
@@ -75,6 +75,52 @@ message Dispatch {
 
   // The dispatch payload
   google.protobuf.Struct payload = 11;
+
+  // The recurrence rule
+  RecurrenceRule recurrence_rule = 12;
+}
+
+// Information about the schedule for reoccurring dispatches
+message RecurrenceRule {
+  // The frequency at which this dispatch should reoccur
+  enum Frequency {
+    FREQUENCY_UNSPECIFIED = 0;
+    FREQUENCY_HOURLY = 1;
+    FREQUENCY_DAILY = 2;
+    FREQUENCY_WEEKLY = 3;
+    FREQUENCY_MONTHLY = 4;
+    FREQUENCY_YEARLY = 5;
+  }
+
+  // The frequency of this recurrence
+  Frequency frequency = 1;
+
+  // The interval between recurrences (e.g. how many hours)
+  uint32 interval = 2;
+
+  // The number of occurences
+  uint32 count = 3;
+
+  // The end timestamp at which this will stop reoccurring
+  google.protobuf.Timestamp until = 4;
+
+  // Hour of the day
+  repeated uint32 by_hours = 5;
+
+  // Day of the week
+  repeated uint32 by_days = 6;
+
+  // Day of the month
+  repeated uint32 by_monthdays = 7;
+
+  // Day of the year
+  repeated uint32 by_yeardays = 8;
+
+  // Week number of the year
+  repeated uint32 by_weeknos = 9;
+
+  // Month of hte year
+  repeated uint32 by_monthnos = 10;
```

Comment:
These look weird IMHO. At first I thought `by_weeknos` was a typo (when I saw the same typo twice I reconsidered :P).

What about `by_xxx_days` and `by_xxx_numbers`?

Or maybe even being much more explicit, I still find the above a bit confusing, and the comment very clear, so:

* `by_hour_of_day`
* `by_day_of_week`
* `by_day_of_month`
* `by_day_of_year`
* `by_week_of_year`
* `by_month_of_year`

While I'm asking questions, why the `by_` prefix?